R/segments.R

```diff
@@ -168,8 +168,9 @@ select_nonintersecting_lines <- function(lines, corridor) {
   } else {
     # Identify the line with maximum number of intersections
     intersecting_lines <- intersections[["origins"]]
-    num_intersections <- sapply(seq_len(length(lines)),
-                                \(x) sum(unlist(intersecting_lines) == x))
+    num_intersections <- vapply(seq_len(length(lines)),
+                                \(x) sum(unlist(intersecting_lines) == x),
+                                numeric(1))
```

Comment:
Shouldn't this return an integer?
```suggestion
                                integer(1))
```